perfaware/sim86/shared/Sim8086.cpp

```diff
@@ -0,0 +1,147 @@
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <string>
+#include <sstream>
+#include "sim86_shared.h"
+#pragma comment (lib, "sim86_shared_debug.lib")
+#define ArrayCount(Array) (sizeof(Array) / sizeof((Array)[0]))
+
+constexpr int RegisterNumber = 15;
+
+s16 Registers[RegisterNumber] = {};
+
+char const* RegisterNames[][3] =
+{
+    {"", "", ""},
+    {"al", "ah", "ax"},
+    {"bl", "bh", "bx"},
+    {"cl", "ch", "cx"},
+    {"dl", "dh", "dx"},
+    {"sp", "sp", "sp"},
+    {"bp", "bp", "bp"},
+    {"si", "si", "si"},
+    {"di", "di", "di"},
+    {"es", "es", "es"},
+    {"cs", "cs", "cs"},
+    {"ss", "ss", "ss"},
+    {"ds", "ds", "ds"},
+    {"ip", "ip", "ip"},
+    {"flags", "flags", "flags"}
+};
+
+void SimulateInstruction(instruction Instruction)
```

Comment:
You can pass the instruction by const reference to avoid a copy.

Reply:
Done d974397bbf5a6f0ba2fb3d73f8095de39fa7bf47